code/modules/clothing/glasses/tajblind.dm

```diff
@@ -0,0 +1,191 @@
+
+	if(mode == MODE_OFF)
+		tint = TINT_BLIND
+		flash_protect = FLASH_PROTECTION_WELDER
+		color_view = null
+		correct_wires = FALSE
+
+	if(mode == MODE_NATURAL)
+		tint = initial(tint)
+		flash_protect = initial(flash_protect)
+		color_view = null
+
+	if(mode == MODE_CORRECTION)
+		tint = initial(tint)
+		flash_protect = initial(flash_protect)
+		color_view = MATRIX_STANDARD
+		correct_wires = TRUE
+
+	if(ishuman(H))
+		if(H.glasses == src)
+			H.update_sight()
+			H.update_client_colour()
+
+/obj/item/clothing/glasses/hud/tajblind/examine(mob/user)
+	. = ..()
+	. += "<span class='info'>Can switch between three modes: Sight-blocking veiled mode, transparent natural sight mode and colorblindness correction mode.</span>"
+
+/obj/item/clothing/glasses/hud/tajblind/examine_more(mob/user)
+	. = ..()
+	. += "Tajaran Veils have long been an important part of their spirituality and culture, suppressed by the Overseers and making a strong return after the civil war. Tajaran believe that to see one’s eyes is to see their soul, and thus the more spiritual Tajara wear veils to conceal their eyes from everyone but the ones closest to them. <br>\
+			These current designs are adapted from recreations of the ancient veils, created by the Alchemists Guild. Technologically advanced and created to help Tajara adapt to life in the larger galactic community, they have systems built-in that allow them to have holographic huds, as well as corrective technology to help Tajaran overcome their genetic tritanopia colour blindness. <br>\
+			Availability on the wider market is highly restricted as a result of their cultural importance, as well as the patent held by the Alchemists Guild, and the lenses are very hard to reverse engineer. Popular theories suggest this as a result of the unique materials available on Adhomai, or the inability to recreate the light conditions of the Tajara homeworld."
+
+/obj/item/clothing/glasses/hud/tajblind/meson
+	name = "\improper Tajaran engineering meson veil"
+	icon_state = "tajblind_engi"
+	item_state = "tajblind_engi"
+
+/obj/item/clothing/glasses/hud/tajblind/meson/Initialize()
+	..()
+	desc += " It has an optical meson scanner integrated into it."
+
+/obj/item/clothing/glasses/hud/tajblind/meson/equipped(mob/user, slot, initial)
+	. = ..()
+	if(slot == SLOT_HUD_GLASSES)
+		ADD_TRAIT(user, TRAIT_MESON_VISION, "meson_glasses[UID()]")
+
+/obj/item/clothing/glasses/hud/tajblind/meson/dropped(mob/user)
+	. = ..()
+	if(user)
+		REMOVE_TRAIT(user, TRAIT_MESON_VISION, "meson_glasses[UID()]")
+
+/obj/item/clothing/glasses/hud/tajblind/meson/cargo
+	name = "\improper Tajaran mining meson veil"
+	icon_state = "tajblind_cargo"
+	item_state = "tajblind_cargo"
+
+/obj/item/clothing/glasses/hud/tajblind/sci
+	name = "\improper Tajaran scientific veil"
+	icon_state = "tajblind_sci"
+	item_state = "tajblind_sci"
+	scan_reagents = 1
+	actions_types = list(/datum/action/item_action/toggle, /datum/action/item_action/toggle_research_scanner)
```

Comment:
im pretty sure this will need `item_action_slot_check` with it just so you cant use the research scanner while its in your pockets